tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.Framework.cs

```diff
@@ -125,9 +191,77 @@ private bool WriteAndEndResponse()
         httpResponse.Flush();
         httpResponse.Close();
         _context.ApplicationInstance.CompleteRequest();
+
         return true;
     }
 
+    private bool TryCreateThrowHttpResponseExceptionDynMeth(out Action<string, string> throwHttpResponseException)
+    {
+        throwHttpResponseException = null;
+
+        try
+        {
+            var exceptionType = Type.GetType("System.Web.Http.HttpResponseException, System.Web.Http");
```

Comment:
I didn't know whether it was best to load the type like this or use `AppDomain.Current`  to search the loaded types.

It seems two different types of assembly names are need because `System.Web.Http` is deployed via nuget so normally in the app's directory where as `System.Net.Http` in deployed with the framework so in the GAC.

Reply:
Another option you have is one we use with other integrations, which is to leverage generics to get the currently loaded assembly: 

https://github.com/DataDog/dd-trace-dotnet/blob/afc371faced8e5342fbf8216bacc86744b58457c/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/CachedMessageHeadersHelper.cs

I'm not sure if it's going to be possible here, but you might be able to do it with a bit of refactoring, and would mean you're not embedding specific assembly versions in there..

Reply:
It seems to me that trick can be used to find the assembly object of the assembly being instrumented, which isn't what we want here.